java/src/main/java/ai/onnxruntime/ONNXEnvironment.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright © 2019, Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime;
+
+import ai.onnxruntime.ONNXSession.SessionOptions;
+
+import java.io.IOException;
+
+/**
+ * The host object for an ONNX system. Can create {@link ONNXSession}s
+ * which encapsulate specific models.
+ */
+public class ONNXEnvironment implements AutoCloseable {
```

Comment:
Do we need to expose Environment to User? Is it a VM wide singleton? Currently there is no use case where user needs to create multiple Environments in a single process. Is there such case in Java?

Reply:
Maybe not. There needs to be a mechanism for creating one with a specific logging level, so a singleton isn't right as it can't be reconstructed (as it might leave dangling sessions and I don't know what the C API will do with those).

I guess I could do something like reference counting and only allow the user to change the env if there are no open sessions. That sounds like it's asking for synchronization bugs, but it might be possible.

Reply:
Environment is now a singleton via reference counting.